### PR TITLE
Deprecate qml.qchem.jordan_wigner

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**
Prior to introduction of the Fermi module in v0.31, there was a `qml.qchem.jordan_wigner` function that took a fermionic operator, defined as list of indices plus notation type, either "chemist" or "physicist" (indicating which indices were lowering vs raising operators). It was not available top-level.

There is now a `qml.jordan_wigner` function in the fermi module. It is available top level. It takes both the old input format used for `qml.qchem.jordan_wigner` and the new input format in terms of fermionic operators from the fermi module.

The plan is to phase out both the `qml.qchem.jordan_wigner` function and the old input format for the function.

**Description of the Change:**
This PR adds a deprecation warning if accessing the `jordan_wigner` function through qchem instead of through the top-level import, so that we can remove it from the qchem module in the future.

It also adds a deprecation warning if using the new `qml.jordan_wigner` in the fermi module with the old input format (list).

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-41449]

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -55,6 +55,16 @@ Pending deprecations
...
 
+* ``qml.qchem.jordan_wigner`` is deprecated, and usage will now raise a warning.
+  Use ``qml.jordan_wigner`` instead. List input to define the fermionic operator
+  is also deprecated; the fermionic operators ``qml.FermiA``, ``qml.FermiC``,
+  ``qml.FermiWord`` and ``qml.FermiSentence`` should be used instead. See the
+  :mod:`pennylane.fermi` module documentation and the
+  `Fermionic Operator <https://pennylane.ai/qml/demos/tutorial_fermionic_operators>`_
+  tutorial for more details.
+
+  - Deprecated in v0.32
```
Comment (trbromley) [inline doc/development/deprecations.rst:66]: For this one, I think it's ok to commit to a removal target. @soranjh do you have a feel for how sensitive this change might be for users? Would one release be enough of warning, or would two be better? I'm on board with 0.33 for removal.

Comment (soranjh) [reply doc/development/deprecations.rst:66]: We have a powerful replacement for this function, so I think one release will be enough.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -39,6 +39,11 @@
 
 <h3>Deprecations 👋</h3>
 
+* ``qml.qchem.jordan_wigner`` is deprecated, use ``qml.jordan_wigner`` instead. 
+  List input to define the fermionic operator is also deprecated; the fermionic 
+  operators ``qml.fermi`` module should be used instead.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:44]:
```suggestion
  operators in the ``qml.fermi`` module should be used instead.
```

**Review by mudit2812 — APPROVED** [review]

**Review by timmysilv — APPROVED** [review]

**Pull request merged**